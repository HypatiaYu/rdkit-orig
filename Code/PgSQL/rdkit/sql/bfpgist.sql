CREATE INDEX fpidx ON pgbfp USING gist (f);

SET rdkit.tanimoto_threshold = 0.7;
SET rdkit.dice_threshold = 0.8;


SET enable_indexscan=off;
SET enable_bitmapscan=off;
SET enable_seqscan=on;

SELECT
    id, tanimoto_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) % f
ORDER BY sml DESC, id limit 10;

SELECT
    id, dice_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) # f
ORDER BY sml DESC, id;

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=off;

SELECT
    id, tanimoto_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) % f
ORDER BY sml DESC, id limit 10;

SELECT
    id, dice_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) # f
ORDER BY sml DESC, id limit 10;

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=on;

DROP INDEX fpidx;
