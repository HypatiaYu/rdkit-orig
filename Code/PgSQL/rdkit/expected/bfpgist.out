CREATE INDEX fpidx ON pgbfp USING gist (f);
SET rdkit.tanimoto_threshold = 0.7;
SET rdkit.dice_threshold = 0.8;
SET enable_indexscan=off;
SET enable_bitmapscan=off;
SET enable_seqscan=on;
SELECT
    id, tanimoto_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) % f
ORDER BY sml DESC, id limit 10;
    id    |        sml        
----------+-------------------
 10560368 | 0.924623115577889
  4287001 | 0.819512195121951
   902176 |             0.785
  1356857 | 0.758620689655172
   690546 | 0.755364806866953
 24761377 | 0.741666666666667
  4440220 |            0.7375
  1941952 | 0.734439834024896
  3218145 | 0.731404958677686
  5161158 | 0.728110599078341
(10 rows)

SELECT
    id, dice_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) # f
ORDER BY sml DESC, id limit 10;
    id    |        sml        
----------+-------------------
 10560368 | 0.960835509138381
  4287001 | 0.900804289544236
   902176 | 0.879551820728291
  1356857 | 0.862745098039216
   690546 | 0.860635696821516
 24761377 | 0.851674641148325
  4440220 | 0.848920863309353
  1941952 |  0.84688995215311
  3218145 | 0.844868735083532
  5161158 | 0.842666666666667
(10 rows)

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=off;
SELECT
    id, tanimoto_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) % f
ORDER BY sml DESC, id limit 10;
    id    |        sml        
----------+-------------------
 10560368 | 0.924623115577889
  4287001 | 0.819512195121951
   902176 |             0.785
  1356857 | 0.758620689655172
   690546 | 0.755364806866953
 24761377 | 0.741666666666667
  4440220 |            0.7375
  1941952 | 0.734439834024896
  3218145 | 0.731404958677686
  5161158 | 0.728110599078341
(10 rows)

SELECT
    id, dice_sml(layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol), f) AS sml
FROM
	pgbfp
WHERE layered_fp('O=C1CC(OC2=CC=CC=C12)C1=CC=CC=C1'::mol) # f
ORDER BY sml DESC, id limit 10;
    id    |        sml        
----------+-------------------
 10560368 | 0.960835509138381
  4287001 | 0.900804289544236
   902176 | 0.879551820728291
  1356857 | 0.862745098039216
   690546 | 0.860635696821516
 24761377 | 0.851674641148325
  4440220 | 0.848920863309353
  1941952 |  0.84688995215311
  3218145 | 0.844868735083532
  5161158 | 0.842666666666667
(10 rows)

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=on;
DROP INDEX fpidx;
